lib/wibox/widget/background.lua

```diff
@@ -21,8 +22,17 @@ function background:draw(context, cr, width, height)
         return
     end
 
+    -- Keep the shape path in case there is a border
+    self._path = nil
+
     if self._shape then
-        self._shape(cr, width, height, unpack(self._shape_args or {}))
+        -- Only add the offset if there is something to draw
+        local offset = ((self._shape_border_width and self._shape_border_color)
+            and self._shape_border_width or 0) / 2
+
+        cr:translate(offset, offset)
+        self._shape(cr, width - 2*offset, height - 2*offset, unpack(self._shape_args or {}))
```

Comment:
Does this work with whatever Lua version has `table.unpack` instead of just `unpack`?

I'll do some experiments with this shape stuff to see what exactly it does. The idea is to print `self.path` and check that the translation works correctly. Hopefully I will find the time to do this on the weekend.